src/bech32.cpp

```diff
@@ -241,7 +241,7 @@ constexpr std::array<uint32_t, 25> GenerateSyndromeConstants() {
     std::array<uint32_t, 25> SYNDROME_CONSTS{};
     for (int k = 1; k < 6; ++k) {
         for (int shift = 0; shift < 5; ++shift) {
-            int16_t b = GF1024_LOG.at(1 << shift);
+            int16_t b = GF1024_LOG.at(1ULL << shift);
```

Comment:
It seems msvc violates the c++ standard by incorrectly promoting this to 64 bit

Reply:
The result of shift operation is promoted to `size_t` as [`std::array::at()`](https://en.cppreference.com/w/cpp/container/array/at) expects an argument of such a type.